patches/roslyn/0004-Fix-the-Roslyn-restore.patch

```diff
@@ -0,0 +1,340 @@
+From 59cec0ae3ffd8b3129e5db5bddce74c1eec297c8 Mon Sep 17 00:00:00 2001
+From: Chris Rummel <crummel@microsoft.com>
+Date: Fri, 22 Sep 2017 14:58:18 -0500
+Subject: [PATCH] Fix the Roslyn restore. Need to condition the VSSDK import on
+ the file existing.
+
+---
+ NuGet.Config                                       |  5 +-
+ build/Targets/Settings.props                       |  3 +-
+ .../Microsoft.CodeAnalysis.Build.Tasks.nuspec      | 45 ++++++------
+ src/NuGet/Microsoft.CodeAnalysis.CSharp.nuspec     |  5 +-
+ src/NuGet/Microsoft.CodeAnalysis.Common.nuspec     | 84 +++++++++++-----------
+ src/NuGet/Microsoft.CodeAnalysis.Compilers.nuspec  |  5 +-
+ .../Microsoft.CodeAnalysis.VisualBasic.nuspec      |  5 +-
+ src/NuGet/Microsoft.Net.Compilers.netcore.nuspec   | 10 +--
+ 8 files changed, 78 insertions(+), 84 deletions(-)
+
+diff --git a/NuGet.Config b/NuGet.Config
+index 0ea90a3391..028616d733 100644
+--- a/NuGet.Config
++++ b/NuGet.Config
+@@ -12,6 +12,7 @@
+   </solution>
+   <packageSources>
+     <clear />
++    <add key="source-built" value="E:\code\source-build\bin/obj/x64/Release/source-built/" />
+     <add key="dotnet.myget.org dotnet-coreclr" value="https://dotnet.myget.org/F/dotnet-coreclr/api/v3/index.json" />
+     <add key="dotnet.myget.org dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
+     <add key="dotnet.myget.org dotnet-corefxtestdata" value="https://dotnet.myget.org/F/dotnet-corefxtestdata/api/v3/index.json" />
+@@ -24,9 +25,9 @@
+     <add key="dotnet.myget.org interactive-window" value="https://dotnet.myget.org/F/interactive-window/api/v3/index.json" />
+     <add key="dotnet.myget.org roslyn-master-nightly" value="https://dotnet.myget.org/F/roslyn-master-nightly/api/v3/index.json" />
+     <add key="dotnet.myget.org devcore" value="https://www.myget.org/F/vs-devcore/api/v3/index.json" />
+-    <add key="dotnet.myget.org roslyn-tools" value = "https://dotnet.myget.org/F/roslyn-tools/api/v3/index.json" />
```

Comment:
This seems like something that should be fixed upstream rather than in the patch.